### PR TITLE
psr-0 and psr-1 compliance

### LINKED ISSUE #14: psr-0 standard compliance
For better integration with other libraries, using the [psr-0](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-0.md) standard would be really usefull.
It would also simplify development and testing a lot, by giving a more organized directory structure.

I'm currently working on improving the library, by adding new features, like arbitrary quantifiers in the pcre form (`{2,4}`, `{4,}`, etc...), case insensitive literals (globally or case by case), unicode modifiers, etc...
But having all these classes sticked into one big file is really slowing down the process.

Would you like me to implement this ?
Proposed directory layout:

```
+ hafriedlander
 \-+ Peg
   |-- Compiler.php
   |-+ Compiler
   | |-- Builder.php
   | |-- Flags.php
   | \-- Writer.php
   |-+ Exception
   | |-- GrammarException.php
   | \-- ParseException.php
   |-- Parser.php
   |-+ Parser
   | |-- ConservativePackrat.php
   | |-- FalseOnlyPackrat.php
   | |-- Packrat.php
   | \-- Regex.php
   |-- PendingRule.php
   |-- Rule.php
   |-- RuleSet.php
   |-- Token.php
   \-+ Token
     |-- Expressionable.php
     |-- ExpressionedRecurse.php
     |-- Literal.php
     |-- Option.php
     |-- Recurse.php
     |-- Regex.php
     |-- Sequence.php
     |-- Terminal.php
     \-- Whitespace.php
```

### PR DESCRIPTION
**Pull request by ju1ius**
closes gh-14

### CODE REVIEW

File: lib/hafriedlander/Peg/Rule.php [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+ *
+ * A rule is the basic unit of a PEG.
+ * This parses one rule, and generates a function that will match on a string
+ *
+ * @author Hamish Friedlander
+ */
+class Rule extends Writer
+{
+	const RULE_RX = '@
+	  (?<name> [\w-]+)                      # The name of the rule
```
Comment (ju1ius) [inline lib/hafriedlander/Peg/Rule.php:22]: This fixes issue gh-2.
See pull request #20

File: lib/hafriedlander/Peg/Rule.php [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+				$active_function = $func_match[1];
+				$this->functions[$active_function] = $func_match[2] . PHP_EOL;
+			}
+			else $this->functions[$active_function] .= $line . PHP_EOL;
+		}
+	}
+
+	/* Manual parsing, because we can't bootstrap ourselves yet */
+  public function parse_rule()
+  {
```
Comment (ju1ius) [inline lib/hafriedlander/Peg/Rule.php:150]: Removed some dead code.
Fixes gh-3

File: lib/vendor/Symfony/Component/ClassLoader/ApcClassLoader.php [hunk 3]
```diff
@@ -0,0 +1,118 @@
+<?php
```
Comment (ju1ius) [inline lib/vendor/Symfony/Component/ClassLoader/ApcClassLoader.php:1]: Symfony ClassLoader component added for ease of use.
Maybe should become a submodule ?

**Comment by hafriedlander** [comment]
Rewrote this to be structured slightly differently after playing with it for a bit. Now in master, so closing this pull request. Thanks.

**Pull request closed without merging**